### PR TITLE
[FLINK-31799][docs] Python connector download link should refer to the url defined in externalized repository

### PR DESCRIPTION
**Pull request by reswqa**
## What is the purpose of the change

*Each externalized connector has its own yml file to manage the url of uber jar. We should introduce a new shortcode to support this.*

## Brief change log
- *Introduce `py_connector_download_link shortcode` for externalized connectors.*

## Verifying this change

Manually verified

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Review by TanYuxin-tyx — APPROVED** [review]
Thanks @reswqa for adding the doc. I only left a minor comment(not strong).

LGTM.

File: docs/layouts/shortcodes/py_connector_download_link.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        {{ end }}
+    </tr>
+    {{ end }}
+    </tbody>
+</table>
+{{ if eq .Site.Language.Lang "en" }}
+See <a href="{{.Site.BaseURL}}{{.Site.LanguagePrefix}}/docs/dev/python/dependency_management/#jar-dependencies">Python dependency management</a>
+for more details on how to use JARs in PyFlink.
+{{ else if eq .Site.Language.Lang "zh" }}
+在 PyFlink 中如何添加 JAR 包依赖参见 <a href="{{.Site.BaseURL}}{{.Site.LanguagePrefix}}/docs/dev/python/dependency_management/#jar-dependencies">Python 依赖管理</a>。
```
[the remark this answers is not in the document]
Comment (reswqa) [reply docs/layouts/shortcodes/py_connector_download_link.html]: Agree with you, `请参考` is more appropriate here.

**Comment by reswqa** [comment]
Thanks @TanYuxin-tyx for the quick review, I have updated this PR according to your comment.

**Comment by TanYuxin-tyx** [comment]
OK, LGTM after the fixing.

**Review by TanYuxin-tyx — APPROVED** [review]

**Pull request merged**